src/background/cookie-instrument.ts

```diff
@@ -1,87 +1,100 @@
-        update.creationTime = creationTime.toLocaleFormat("%Y-%m-%d %H:%M:%S");
+    // Instrument cookie changes
+    browser.cookies.onChanged.addListener(
+      async (changeInfo: {
+        /** True if a cookie was removed. */
+        removed: boolean;
+        /** Information about the cookie that was set or removed. */
+        cookie: Cookie;
+        /** The underlying reason behind the cookie's change. */
+        cause: OnChangedCause;
+      }) => {
+        const change = changeInfo.removed ? "deleted" : "added";
 
-        // Expiry time (in seconds)
-        // May return ~Max(int64). I believe this is a session
-        // cookie which doesn't expire. Sessions cookies with
-        // non-max expiry time expire after session or at expiry.
-        const expiryTime = cookie.expiry; // returns seconds
-        if (expiryTime === 9223372036854776000) {
-          const expiryTimeString = "9999-12-31 23:59:59";
-        } else {
-          const expiryTimeDate = new Date(expiryTime * 1000); // requires milliseconds
-          const expiryTimeString = expiryTimeDate.toLocaleFormat(
-            "%Y-%m-%d %H:%M:%S",
-          );
-        }
-        update.expiry = expiryTimeString;
-        update.is_http_only = loggingDB.boolToInt(cookie.isHttpOnly);
-        update.is_session = loggingDB.boolToInt(cookie.isSession);
+        // TODO: Support other cookie operations
+        if (
+          change === "deleted" ||
+          change === "added" ||
+          change === "changed"
+        ) {
+          const update = {} as JavascriptCookieChange;
+
+          update.change = change;
+          // TODO: Add changeInfo.cause
+          update.crawl_id = crawlID;
+
+          const cookie: Cookie = changeInfo.cookie;
 
-        // Accessed time (in microseconds)
-        const lastAccessedTime = new Date(cookie.lastAccessed / 1000); // requires milliseconds
-        update.last_accessed = lastAccessedTime.toLocaleFormat(
-          "%Y-%m-%d %H:%M:%S",
-        );
-        update.raw_host = loggingDB.escapeString(cookie.rawHost);
+          // Creation time (in microseconds)
+          // const creationTime = new Date(cookie.creationTime / 1000); // requires milliseconds
+          // update.creationTime = creationTime.toLocaleFormat("%Y-%m-%d %H:%M:%S");
 
-        cookie = cookie.QueryInterface(Ci.nsICookie);
-        update.expires = cookie.expires;
-        update.host = loggingDB.escapeString(cookie.host);
-        update.is_domain = loggingDB.boolToInt(cookie.isDomain);
-        update.is_secure = loggingDB.boolToInt(cookie.isSecure);
-        update.name = loggingDB.escapeString(cookie.name);
-        update.path = loggingDB.escapeString(cookie.path);
-        update.policy = cookie.policy;
-        update.status = cookie.status;
-        update.value = loggingDB.escapeString(cookie.value);
+          // Expiry time (in seconds)
+          // A comment from pre-webextension code, which may still be valid:
```

Comment:
This would be nice to test, hopefully the info exposed through webextensions is cleaner.